client/client.go

```diff
@@ -45,12 +40,29 @@ func (c *client) Query(ctx context.Context, query interface{}, variables map[str
 	}
 
 	err = apiClient.Query(ctx, query, variables, opts...)
+	return c.determineClientError(ctx, apiClient, opts, err)
+}
+
+func (c *client) determineClientError(ctx context.Context, client *graphql.Client, opts []graphql.RequestOption, err error) error {
+	if err == nil {
+		return nil
+	}
 
-	if err != nil && err.Error() == "unauthorized" {
-		return fmt.Errorf("unauthorized: you can re-login using `spacectl profile login`")
+	if !strings.Contains(err.Error(), "unauthorized") {
+		return err
+	}
+
+	var query struct {
+		Viewer *struct {
+			ID string `graphql:"id" json:"id"`
+		}
+	}
+	queryErr := client.Query(ctx, &query, map[string]interface{}{}, opts...)
```

Comment:
I understand that this isn't very DRY as we're almost 100% repeating ourselves here, however I don't want to deal with crazy import cycles or nonsensical packages. This is good.